assets/js/modules/analytics/components/module/ModuleOverviewWidget/index.js

```diff
@@ -74,29 +75,42 @@ export default function ModuleOverviewWidget( { Widget, WidgetReportError } ) {
 	const statsReport = useSelect( ( select ) => select( MODULES_ANALYTICS ).getReport( statsArgs ) );
 	const statsError = useSelect( ( select ) => select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ statsArgs ] ) );
 
-	if ( overviewError || statsError ) {
-		return <WidgetReportError statsError={ overviewError || statsError } />;
-	}
+	const isZero = isZeroReport( overviewReport );
+	const isError = overviewError || statsError;
 
 	return (
-		<Widget Header={ Header } noPadding>
-			<Overview
-				loaded={ overviewLoaded }
-				report={ overviewReport }
-				selectedStat={ selectedStat }
-				handleStatSelection={ setSelectedState }
-			/>
+		<Widget
+			Header={ Header }
+			noPadding={ ! isError && ! isZero }
+		>
+			{ isError && (
+				<WidgetReportError moduleSlug="analytics" error={ overviewError || statsError } />
+			) }
+			{ ( ! isError && isZero ) && (
+				<WidgetReportZero moduleSlug="analytics" />
+			) }
+			{ ( ! isError && ! isZero ) && (
```

Comment:
Nitpick: Instead of having all these conditions, I think the code would be easier to follow if it was using early returns, the same way that you've adjusted it in `ModulePopularPagesWidget`:
* If `isError`, return `WidgetReportError` wrapped in `Widget` with the `Header`.
* If `isZero`, return `WidgetReportZero` wrapped in `Widget` with the `Header`.
* Otherwise render the default content with `noPadding` simply set to `true`.

Reply:
Yep, updated!